sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -463,25 +461,26 @@ def prepare_data_for_lloyd(X_t, init, tol, sample_weight, copy_x):
     max_work_group_size = device.max_work_group_size
 
     sum_axis1_kernel = make_sum_reduction_2d_kernel(
-        X_t.shape[0],
-        X_t.shape[1],
+        X_t.shape,
         axis=1,
         work_group_size="max",
         device=device,
         dtype=compute_dtype,
     )
 
-    elementwise_binary_divide_kernel = make_elementwise_binary_op_1d_kernel(
-        n_features, _divide, max_work_group_size, compute_dtype
+    divide_by_n_samples_kernel = make_apply_elementwise_func(
+        (n_features,),
+        _divide_by(compute_dtype(n_samples)),
```

Comment:
I do not find this chaining super intuitive. Can you extract it above just above the call to this function and give it a name? :slightly_smiling_face: